### ISSUE TITLE
Bad Request: while fetching all the files within a bucket

### ISSUE THREAD

**Issue by rudrankriyam** [satishbabariya/storage-swift#2]
# Bug report

## Describe the bug

Bad request error while fetching all the files within a bucket. 

## To Reproduce

After fetching the buckets successfully, the list method returns a bad request. 

```swift 
let client = SupabaseClient(supabaseUrl: "{ base url }", supabaseKey: "{ key }")

client.storage.listBuckets { result in
    switch result {
        case .success(let buckets):
            guard let bucket = buckets.first else { return }
            let file = client.storage.from(id: bucket.id)
            
            file.list { fileResult in
                switch fileResult {
                    case .success(let objects):
                        print(objects)
                    case .failure(let error):
                        print(error)
                }
            }
        case .failure(let error):
            print(error)
    }
}
```

## Expected behaviour

After forking the package and printing the data, I found that the body is missing the prefix parameter - 

```
body should have required property 'prefix'
```

Adding the prefix parameter worked, and it returned the files. However, there's no mention of the prefix in the JS [docs](https://supabase.io/docs/reference/javascript/storage-from-list).

**Comment by CodeJonesW** [comment]
I am having a similar issue when using [from.remove()](https://supabase.io/docs/reference/javascript/storage-from-remove#delete-file) 
message: "body should have required property 'prefixes'", status: 400

How did you add this property using the client methods?

**Comment by rudrankriyam** [comment]
I forked the repo, added the package to my project and made the change manually. I couldn’t think of any alternate solutions

**Comment by CodeJonesW** [comment]
Thanks

**Comment by rudrankriyam** [comment]
Their [C#](https://github.com/supabase-community/supabase-csharp/blob/41c01ff8f93cbd6cf607fde0e81ffab54ef421ef/Supabase/Storage/StorageFileApi.cs#L253) and python clients do use prefixes in the body, so it looks like the documentation is outdated. 

You can fork and make the necessary changes. I'll send a PR for this

**Comment by rudrankriyam** [comment]
Solved by #3
